core/src/main/java/nu/yona/server/subscriptions/service/UserService.java

```diff
@@ -247,27 +252,33 @@ private void verifyUserDoesNotExist(String mobileNumber)
 		throw UserServiceException.userExists(mobileNumber);
 	}
 
-	private void verifyUserIsAllowed(String mobileNumber)
+	private void verifyUserIsAllowed(String mobileNumber, UserSignUp origin)
+	{
+		verifyBelowMaxUsers();
+		verifyWhiteList(mobileNumber, origin);
+	}
+
+	private void verifyBelowMaxUsers()
 	{
 		int maxUsers = yonaProperties.getMaxUsers();
-		if (maxUsers <= 0)
```

Comment:
Indeed. There is no value in allowing negative values. If you don't like to use this property, just set it to a very large value (as we do today).